internal/venice-common/src/main/java/com/linkedin/venice/meta/StoreInfo.java

```diff
@@ -416,6 +418,20 @@ public Optional<Version> getVersion(int versionNum) {
     return Optional.empty();
   }
 
+  /**
+   * Get backup version number, the largest online version number that is less than the current version number
+   */
+  public int getBackupVersionNumber() {
+    int currentVersion = getCurrentVersion();
+    getVersions().sort(Comparator.comparingInt(Version::getNumber).reversed());
```

Comment:
This line of sorting logic is not necessary?

Reply:
Why is it not needed? Could we guarantee that version list in store info is always sorted by number?

Reply:
Never mind. I mis-understand the logic here, you are trying to loop from higher version to lower verison to find the newest online version.